tests/tests_geomstats/test_complex_matrices.py

```diff
@@ -63,170 +43,30 @@ def test_frobenius_product(self, mat_a, mat_b, expected):
             gs.array(expected),
         )
 
-    def test_trace_product(self, mat_a, mat_b, expected):
-        self.assertAllClose(
-            ComplexMatrices.trace_product(
-                gs.array(mat_a, dtype=CDTYPE),
-                gs.array(mat_b, dtype=CDTYPE),
-            ),
-            gs.array(expected),
-        )
-
-    def test_flatten(self, m, n, mat, expected):
-        self.assertAllClose(
-            ComplexMatrices(m, n).flatten(gs.array(mat, dtype=CDTYPE)),
-            gs.array(expected),
-        )
-
-    def test_transpose(self, m, n, mat, expected):
-        self.assertAllClose(
-            ComplexMatrices(m, n).transpose(gs.array(mat, dtype=CDTYPE)),
-            gs.array(expected),
-        )
-
     def test_transconjugate(self, m, n, mat, expected):
         self.assertAllClose(
             ComplexMatrices(m, n).transconjugate(gs.array(mat, dtype=CDTYPE)),
             gs.array(expected),
         )
 
-    def test_diagonal(self, m, n, mat, expected):
-        self.assertAllClose(
-            ComplexMatrices(m, n).diagonal(gs.array(mat, dtype=CDTYPE)),
-            gs.array(expected),
-        )
-
-    def test_is_diagonal(self, m, n, mat, expected):
-        self.assertAllClose(
-            ComplexMatrices(m, n).is_diagonal(gs.array(mat, dtype=CDTYPE)),
-            expected,
-        )
-
-    def test_is_symmetric(self, m, n, mat, expected):
-        self.assertAllClose(
-            ComplexMatrices(m, n).is_symmetric(gs.array(mat, dtype=CDTYPE)),
-            gs.array(expected),
-        )
-
     def test_is_hermitian(self, m, n, mat, expected):
         self.assertAllClose(
             ComplexMatrices(m, n).is_hermitian(gs.array(mat, dtype=CDTYPE)),
             gs.array(expected),
         )
 
-    def test_is_skew_symmetric(self, m, n, mat, expected):
-        self.assertAllClose(
-            ComplexMatrices(m, n).is_skew_symmetric(gs.array(mat, dtype=CDTYPE)),
-            gs.array(expected),
-        )
-
-    def test_is_pd(self, m, n, mat, expected):
-        self.assertAllClose(
-            ComplexMatrices(m, n).is_pd(gs.array(mat, dtype=CDTYPE)),
-            gs.array(expected),
-        )
-
-    def test_is_spd(self, m, n, mat, expected):
-        self.assertAllClose(
-            ComplexMatrices(m, n).is_spd(gs.array(mat, dtype=CDTYPE)),
-            gs.array(expected),
-        )
-
     def test_is_hpd(self, m, n, mat, expected):
         self.assertAllClose(
             ComplexMatrices(m, n).is_hpd(gs.array(mat, dtype=CDTYPE)),
             gs.array(expected),
         )
 
-    def test_is_upper_triangular(self, m, n, mat, expected):
```

Comment:
We use the methods from `Matrices` to compute on `ComplexMatrices`. Which is great, because it avoids duplicated code. Do we still want to check that these methods work for _complex_ matrices, i.e. for arrays with complex values? 🤔

Reply:
The idea is to move these methods from `Matrices` to the backend and then add complete tests there (i.e. checking also complex if they can handle it). This step towards removing all this methods from `ComplexMatrices` is to make it easier to identify which methods are specific (the ones kept there for now).